train.py

```diff
@@ -3,340 +3,388 @@
+                data_loader=train_set_loader,
+                model=model,
+                training=True,
+                metric_tracker=metric_tracker_train,
+            )
+            run_epoch(
+                epoch,
+                data_loader=val_set_loader,
+                model=model,
+                training=False,
+                metric_tracker=metric_tracker_val,
+            )
             scheduler.step()
 
             metric_tracker_train.plot(
-                path="{}/train/metrics.png".format(images_filepath)
+                path=f"{images_filepath}/train/metrics.png"
             )
-            metric_tracker_test.plot(path="{}/test/metrics.png".format(images_filepath))
+            metric_tracker_val.plot(path=f"{images_filepath}/val/metrics.png")
             metric_tracker_train.save()
-            metric_tracker_test.save()
+            metric_tracker_val.save()
 
             ################################################################################ Saving models
             if args.save:
                 state = {
                     "args": args,
                     "epoch": epoch,
-                    "net": net.state_dict(),
+                    "model": model.state_dict(),
                     "optimizer": optimizer.state_dict(),
                     "scheduler": scheduler.state_dict(),
                 }
+
+                current_epoch_filename = "{}_ckpt.pth.tar".format(epoch)
+                latest_epoch_filename = "latest_ckpt.pth.tar".format(epoch)
+
                 epoch_pbar.set_description(
-                    "Saving at {}/ckpt.pth.tar".format(saved_models_filepath)
+                    "Saving at {}/{}".format(
+                        saved_models_filepath, current_epoch_filename
+                    )
+                )
+
+                save_checkpoint(
+                    state=state,
+                    directory=saved_models_filepath,
+                    filename=current_epoch_filename,
+                    is_best=False,
                 )
-                filename = "ckpt.pth.tar".format(epoch)
-                previous_save = "{}/ckpt.pth.tar".format(
-                    saved_models_filepath, epoch - 1
+
+                epoch_pbar.set_description(
+                    "Saving at {}/{}".format(
+                        saved_models_filepath, latest_epoch_filename
+                    )
                 )
-                if os.path.isfile(previous_save):
-                    os.remove(previous_save)
+
                 save_checkpoint(
                     state=state,
                     directory=saved_models_filepath,
-                    filename=filename,
+                    filename=latest_epoch_filename,
                     is_best=False,
                 )
             ############################################################################################################
 
             epoch_pbar.set_description("")
             epoch_pbar.update(1)
+
```

Comment:
This is fantastic functionality. But this should only run if the test flag is set, right? I don't see a check.

Reply:
Will add one now. :)